client/components/checkout-form.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+export default class CheckoutForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      creditCard: null,
+      shippingAddress: ''
+    };
+    this.setView = this.setView.bind(this);
+    this.handleChangeName = this.handleChangeName.bind(this);
+    this.handleChangeCC = this.handleChangeCC.bind(this);
+    this.handleChangeAddress = this.handleChangeAddress.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+
+  handleChangeName(event) {
+    this.setState({ name: event.target.value });
+  }
+
+  handleChangeCC(event) {
+    this.setState({ creditCard: event.target.value });
+
+  }
+
+  handleChangeAddress(event) {
+    this.setState({ shippingAddress: event.target.value });
+  }
```

Comment:
You should be able to use one `onChange` handler for your inputs. You can give each input a `name` attribute set to the same value as the property name in state then use the name attribute to set the correct property in state. For example your `name` input would look like:

```JSX
<input type="text" name="name" className="form-control" id="name" value={this.state.name} onChange={this.handleChange}/>
```

The your `onChange` can look like this and be used for all the inputs:

```JSX
handleChange(event) {
  const name = event.target.name;
  const value = event.target.value;
  
  this.setState({ 
    [ name ]: value
  });
}
```